### PR TITLE
Move meta-module creation into Spack extension

### PR DESCRIPTION
**Pull request by kgerheiser**
Command becomes `spack stack setup-meta-modules` and then the original code is called directly. Only change was using `ev.active_envirnonment()` to query the environment  and wrapping the main code in a function.

Also, added a kind of hacky unit-test that creates an empty environment and re-creates the necessary directory structure for the meta-module command.

### CODE REVIEW

File: lib/jcsda-emc/spack-stack/stack/cmd/stack.py [hunk 1]
```diff
@@ -21,10 +22,14 @@ def setup_parser(subparser):
     sp = subparser.add_subparsers(metavar='SUBCOMMAND', dest='stack_command')
     create_parser = sp.add_parser('create',
                                   help='Create spack-stack env or container.')
+    meta_modules_parser = sp.add_parser('setup-meta-modules', help='Create LMod meta-modules')
```
Comment (climbfuji) [inline lib/jcsda-emc/spack-stack/stack/cmd/stack.py]:
```suggestion
    meta_modules_parser = sp.add_parser('setup-meta-modules', help='Create lmod/lua or tcl/tk meta-modules')
```

**Review by climbfuji — CHANGES_REQUESTED** [review]
This looks nice, will try out today. For the sake of consistency, what do you think of changing `spack stack env` to `spack stack environment` because we also spell it out for the meta modules and the containers?

**Comment by kgerheiser** [comment]
Sure, we can make it `environment`. Just thought I would save a few letters.

**Comment by climbfuji** [comment]
@kgerheiser Looks good. It's still a draft, please let me know when you think it's ready so that I don't miss it and you have to wait. Thanks!

**Review by climbfuji — APPROVED** [review]
